mpmp/src/cmds/Chat.java

```diff
@@ -19,7 +19,22 @@ public void exec(String line, Conn conn) {
 			c.sendErr(ErrCode.NotSubscribed);
 			return;
 		}
-
+		
+		String[] args = line.split("@");
+		if (args.length > 2) {
+			c.sendErr("Don't put a '@' in your message if you want to whisper, please.");
+			return;
+		}
+		if (args.length > 1) {
+			String name = args[1].substring(0, args[1].indexOf(' '));
```

Comment:
Oh, sorry. I forgot to mention that our user names can have spaces in them (I meant to remind you). So we can't just test for a space. Rather, we must try to match the given message to the longest user name. The code below finds the longest username that is preceded by an `@`, regardless of spaces.

```java
bestmatch = first user-name;
for(<get username>)
    if(msg.matches("@" + username) && username.length() > bestmatch.length())
        bestmatch = username;
```